src/nwm_routing/src/nwm_routing/preprocess.py

```diff
@@ -206,31 +198,38 @@ def nwm_initial_warmstate_preprocess(
                 len(waterbodies_initial_states_df)
             )
 
+        waterbodies_df = pd.merge(
+            waterbodies_df, waterbodies_initial_states_df, on="lake_id"
+        )
+
         if verbose:
             print("waterbody initial states complete")
         if showtiming:
             print("... in %s seconds." % (time.time() - start_time))
             start_time = time.time()
 
-    # STEP 4: Handle Channel Initial States
+    # STEP 4: Handle Channel Initial States and Set T0
     if showtiming:
         start_time = time.time()
     if verbose:
         print("setting channel initial states ...")
 
     q0 = nnu.build_channel_initial_state(restart_parameters, segment_index)
 
+    if restart_parameters.get("wrf_hydro_channel_restart_file",None):
+        channel_initial_states_file = restart_parameters["wrf_hydro_channel_restart_file"]
+        t0_str = nhd_io.get_param_str(channel_initial_states_file, "Restart_Time")
+    else:
+        t0_str = "2015-08-16_00:00:00"
```

Comment:
@jameshalgren I will use this t0. Do you have a method to get t0 for a cold start?

Reply:
In short, no. See #427 for additional detail on updating the t0 read.

Reply:
I actually don't need this t0.